### DISCUSSION TITLE
Is there a way to .skip(10) rows in a query

### DISCUSSION THREAD

**Discussion by AaronNGray** [supabase/supabase#2038] [Questions] [answered]
There is a .limit() in data queries is there a way to .skip() rows like in Knex in a query as a way to implement pagination please ?

To generate a "OFFSET n ROWS" clause.

**Comment by steve-chavez** [comment, marked as answer]
Hey @AaronNGray,

Have you seen the [range](https://supabase.io/docs/reference/javascript/range) modifier? You could implement pagination with that.

**Reply by AaronNGray** [reply]
Not sure that would work for my use case as I am already doing an .lte on a datetime field.

**Comment by AaronNGray** [comment]
Looks like a .skip() could be coded the simular to .limit() :- 

https://github.com/supabase/postgrest-js/blob/653323215571bdf7074303998ad507f5e96f0ada/src/lib/PostgrestTransformBuilder.ts#L66

CanI do an issue then put in a patch and pull request please ?

**Reply by steve-chavez** [reply]
You mean you'd like to add an `skip()` method that underlyingly uses `offset`? Like done [here](https://github.com/supabase/postgrest-js/blob/653323215571bdf7074303998ad507f5e96f0ada/src/lib/PostgrestTransformBuilder.ts#L80)?

(Docs about [offset](https://postgrest.org/en/stable/api.html?highlight=offset#limits-and-pagination))

I think that could work. Perhaps we should call the method just `offset()` though?

**Reply by soedirgo** [reply]
A `range` with a `to` of your table count/a very large number should achieve the same thing here. Can you tell me how the `.lte` makes it not fit your use case?

**Reply by AaronNGray** [reply]
Ah okay I have been barking up the wrong tree @soedirgo I will try ```.range(start,start+count)```
